evap/results/views.py

```diff
@@ -225,10 +226,14 @@ def remove_textanswers_that_the_user_must_not_see(evaluation_result, user, repre
                     if can_textanswer_be_seen_by(user, represented_users, answer, view)
                 ]
         # remove empty TextResults
-        questionnaire_result.question_results = [
-            result for result in questionnaire_result.question_results
-            if not isinstance(result, TextResult) or len(result.answers) > 0
-        ]
+        cleaned_results = []
+        for result in questionnaire_result.question_results:
+            if not isinstance(result, TextResult) or len(result.answers) > 0:
+                cleaned_results.append(result)
+            # if previous result was a RatingResult, update info about additional text result
+            elif cleaned_results and isinstance(cleaned_results[-1], RatingResult):
```

Comment:
it took me a while to understand why this is correct :) how about `elif is_additional_text_result`?